include/envoy/server/overload_manager.h

```diff
@@ -81,6 +95,14 @@ class OverloadManager {
    * an alternative to registering a callback for overload action state changes.
    */
   virtual ThreadLocalOverloadState& getThreadLocalOverloadState() PURE;
+
+  /**
+   * Convenience method to get a statically allocated reference to the inactive overload
+   * action state.
+   */
+  static const OverloadActionState& getInactiveState() {
+    CONSTRUCT_ON_FIRST_USE(OverloadActionState, OverloadActionState::Inactive);
```

Comment:
How come you can't just write `OverloadActionState::Inactive` inline below? I'm guessing the compiler does a reasonable job of optimizing.

Reply:
I need this to return a reference to a statically allocated variable (an lvalue) so that it can be used in the ConnectionManagerImpl constructor for initializing the overload_stop_accepting_requests_ field in the case where the overload_manager is null (this is the case for the HCM used by the admin port since we want that port to be accessible and respond to requests even when envoy is overloaded).

Reply:
Still a bit confused on this one; AFAIK, `enum class` is essentially an integer and can be passed-by-value, why do we use references here?

Reply:
overload_stop_accepting_requests_ is itself a [reference](https://github.com/envoyproxy/envoy/pull/4126/files/76844356295d631265562891b75f4dbf2df8d636#diff-b22ef653a728911bc6fc608f8558ef6eR461), pointing either to an element in the ThreadLocalOverloadState map, or to the memory location returned by this method. I can  add a comment here to explain the use-case better.

Reply:
Ah, I see, yes make sense, thanks.

Reply:
added comment